pkgs/by-name/co/cosh/package.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, stdenv
+, rustPlatform
+, rustc
+, fetchFromGitHub
+, openssl
+, pkg-config
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "cosh";
+  version = "0.1-unstable-2024-01-27";
+  src = fetchFromGitHub {
+    repo = "cosh";
+    owner = "tomhrr";
+    rev = "540d3b99908dc2603887f64e288ac18b7f53cb37";
+    hash = "sha256-O+DnXMFVHgPBQcuOmvtyMKEYYwCC9fFY594pEWgU2HM=";
+  };
+
+  # the tests try to read /etc/localtime which is unavailable under the sandbox
+  doCheck = false;
+
+  nativeBuildInputs = [ pkg-config rustPlatform.bindgenHook ];
+  buildInputs = [ openssl ];
+
+  cargoLock.lockFile = ./Cargo.lock;
+
+  postPatch = ''
+    ln -s ${./Cargo.lock} Cargo.lock
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+    make prefix=$out all
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    make prefix=$out install
```

Comment:
```sh
$ results/cosh/bin/wrapped-cosh
results/cosh/bin/wrapped-cosh: line 5: /usr/local/bin/cosh: No such file or directory
Press enter to continue
```

Reply:
this is a bug in the upstream, not in nix. previously it would simply fail if the dotfiles were not present. i'll try updating via unstableGitUpdater, i think it got fixed.

Reply:
updated via unstableGitUpdater, this should be fixed now.